build/dsc/resource.rb

```diff
@@ -93,26 +93,29 @@ def absentable?
       @absentable
     end
 
-    def absent_value
-      @absent_value ||=
-        if ensure_property.values.any? { |v| v.casecmp('absent') == 0 }
-          'absent'
-        elsif ensure_property.values.any? { |v| v.casecmp('disable') == 0 }
-          'disable'
-        else
-          throw 'Error processing MOF schema - could not determine equivalent \'absent\' value for ensure'
-        end
+    def ensure_hash
+      [
+        { :name => 'MSFT_xRunbookDirectory',      :present => ['published', 'draft'], :absent => [ 'absent'] },
+        { :name => 'MSFT_WindowsOptionalFeature', :present => ['enable'], :absent => [ 'disable'] },
+      ]
     end
 
-    def ensure_value
-      @ensure_value ||=
-        if ensure_property.values.any? { |v| v.casecmp('present') == 0 }
-          'present'
-        elsif ensure_property.values.any? { |v| v.casecmp('enable') == 0 }
-          'enable'
-        else
-          throw 'Error processing MOF schema - could not determine equivalent \'present\' value for ensure'
-        end
+    def valid_ensure_present_values
+      foo = ensure_hash.detect{|v| v[:name] == name }
```

Comment:
Then this can just be `ensure_hash[name]` (unless we want to do case insensitive, in which case I suggest we store lower case versions in the hash and call `name.downcase` here for a key)